plugins/module_utils/aci.py

```diff
@@ -140,18 +142,19 @@ def __init__(self, module):
             self.module.warn('Enable debug output because ANSIBLE_DEBUG was set.')
             self.params['output_level'] = 'debug'
 
-        if self.params.get('private_key'):
-            # Perform signature-based authentication, no need to log on separately
-            if not HAS_CRYPTOGRAPHY and not HAS_OPENSSL:
-                self.module.fail_json(
-                    msg='Cannot use signature-based authentication because cryptography (preferred) or pyopenssl are not available')
-            elif self.params.get('password') is not None:
-                self.module.warn("When doing ACI signatured-based authentication, providing parameter 'password' is not required")
-        elif self.params.get('password'):
-            # Perform password-based authentication, log on using password
-            self.login()
-        else:
-            self.module.fail_json(msg="Either parameter 'password' or 'private_key' is required for authentication")
+        if self.module._socket_path is None:
+            if self.params.get('private_key'):
```

Comment:
This requires to not use the httpapi if you want to use the private key.
Would prefer if we move it to httpapi too or we make it independent.